chall02/mpernia-.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+
+import sys
+
+morse = {
+	'a':'.-',
+	'b':'-...',
+	'c':'-.-.',
+	'd':'-..',
+	'e':'.',
+	'f':'..-.',
+	'g':'--.',
+	'h':'....',
+	'i':'..',
+	'j':'.---',
+	'k':'-.-',
+	'l':'.-..',
+	'm':'--',
+	'n':'-.',
+	'o':'---',
+	'p':'.--.',
+	'q':'--.-',
+	'r':'.-.',
+	's':'...',
+	't':'-',
+	'u':'..-',
+	'v':'...-',
+	'w':'.--',
+	'x':'-..-',
+	'y':'-.--',
+	'z':'--..'
+}
+
+if len(sys.argv) == 2 and all(x.isalpha() or x.isspace() for x in sys.argv[1]) and len(sys.argv[1]) > 0:
```

Comment:
Nice! You are using comprehension!
That's really great!
However here there is a more optimized approach:
```suggestion
if len(sys.argv) == 2 and sys.argv[1].replace(" ", "").isalpha():
```
Note that it also covers then `len(sys.argv[1]) > 0` condition as `"".isalpha()` returns `False`

Reply:
Thank you for the review, that's very helpful. The changes you made blow my mind. Why my file couldn't be executed?

Reply:
I'll answer for him, no execution rights on the file. It had to be executed as `./file.py`, not as `your_python_interpreter file.py`.

Reply:
Uhmm... I understand, but besides chmod there's something else to do? Because, I have this:
`-rwxrwxrwx 1 mmx66 mmx66  663 abr  6 16:19 mpernia-.py`

Reply:
You only need the shebang (which you had), and the execution permissions. At the time of the last commit (826e6e7), your permissions were -rw-r--r-- (644), so it wasn't executable :)

Reply:
Thank you, @abguimba. You were very helpful, :)
I'm trying to understand to send the new one, we'll see if I have luck this time. >:D